core/src/main/java/com/github/mizool/core/password/PasswordHasherRegistry.java

```diff
@@ -31,8 +37,27 @@
 
     public PasswordHasher getHasher(String algorithm)
     {
-        return Iterables.getOnlyElement(
-            Iterables.filter(
-                passwordHashers, passwordHasher -> passwordHasher.isResponsibleFor(algorithm)));
+        return Iterables.getOnlyElement(Iterables.filter(passwordHashers,
+            passwordHasher -> passwordHasher.isResponsibleFor(algorithm)));
+    }
+
+    public static PasswordHasher getDefaultHasher()
+    {
+        Iterable<PasswordHasher> hasherInstances = MetaInfServices.instances(PasswordHasher.class);
+        ImmutableList<PasswordHasher> defaultPasswordHashers = Streams.sequential(hasherInstances)
+            .collect(ImmutableList.toImmutableList());
+
+        Optional<PasswordHasher> defaultHasher = defaultPasswordHashers.stream().findFirst();
+        if (!defaultHasher.isPresent())
+        {
+            throw new CodeInconsistencyException(
+                "No hasher instances for PasswordHasher found, one should be registered with META-INF services");
+        }
+        if (defaultPasswordHashers.size() > 1)
+        {
+            throw new CodeInconsistencyException(
+                "Found several hasher instances for PasswordHasher, only one should be registered with META-INF services");
```

Comment:
This won't work. To my knowledge, applications cannot reliably un-register a service that is registered by a jar file on their classpath.

Let's change this as follows:

1. _All_ hasher implementations register via `@MetaInfServices`
2. Among all those hashers, `getDefaultHasher()` finds the one whose name matches the one given by a system property `com.github.mizool.core.password.PasswordHasherRegistry.defaultAlgorithm`. If it is unset, the registry defaults to `Pbkdf2WithHmacSha512Hasher.ALGORITHM_NAME`.
3. PasswordHashTool always iterates through all hashers and displays each digest (allowing the developer to pick the algorithm in use by their application)